tests/src/test/java/org/eclipse/hono/tests/amqp/AmqpAdapterBase.java

```diff
@@ -0,0 +1,289 @@
+        client.connect(new ProtonClientOptions()).compose(ok -> {
+            clientsToShutdown.add(client);
+            return getOrCreateSender(tenantId, client);
+        }).compose(sender -> {
+            final Message msg = ProtonHelper.message("some payload");
+            return sender.send(msg);
+        }).setHandler(context.asyncAssertFailure(t -> {
+            context.assertTrue(ClientErrorException.class.isInstance(t));
+
+            // HonoClient maps AmqpError.UNAUTHORIZED_ACCESS to HttpURLConnection.HTTP_BAD_REQUEST
+            // -> so we test the message
+
+            final String expectedErrorMsg = String
+                    .format("This adapter is not enabled for tenant [tenantId: %s].", tenantId);
+
+            context.assertEquals(expectedErrorMsg, ((ClientErrorException) t).getMessage());
+
+            completionTracker.complete();
+        }));
+        completionTracker.awaitSuccess();
+    }
+
+    /**
+     * Verifies that a number of messages published through the AMQP adapter can be successfully consumed by
+     * applications connected to the AMQP messaging network.
+     *
+     * @param context The Vert.x test context.
+     * @throws InterruptedException Exception.
+     */
+    @Test
+    public void testUploadingXnumberOfMessages(final TestContext context) throws InterruptedException {
+
+        final String tenantId = helper.getRandomTenantId();
+        final String deviceId = helper.getRandomDeviceId(tenantId);
+        final String username = IntegrationTestSupport.getUsername(deviceId, tenantId);
+        final String password = "verysecret";
+
+        final CountDownLatch received = new CountDownLatch(MESSAGES_TO_SEND);
+
+        final Async addTenantTracker = context.async();
+        final TenantObject tenant = TenantObject.from(tenantId, true);
+        helper.registry.addDeviceForTenant(tenant, deviceId, password)
+                .setHandler(context.asyncAssertSuccess(ok -> addTenantTracker.complete()));
+        addTenantTracker.await();
+
+        final Async setup = context.async();
+
+        final HonoClient client = prepareAmqpAdapterClient(username, password);
+        client.connect(new ProtonClientOptions()).compose(ok -> {
+            clientsToShutdown.add(client);
+            return Future.succeededFuture();
+        }).compose(ok -> createConsumer(tenantId, msg -> {
+            received.countDown();
+            if (received.getCount() % 40 == 0) {
+                LOG.info("message {} received", MESSAGES_TO_SEND - received.getCount());
+            }
+        })).setHandler(context.asyncAssertSuccess(ok -> setup.complete()));
+        setup.await();
+
+        final AtomicInteger messageCount = new AtomicInteger(0);
+
+        while (messageCount.get() < MESSAGES_TO_SEND) {
+            final Async sendingComplete = context.async();
+            getOrCreateSender(tenantId, client).compose(sender -> {
```

Comment:
I do not think that you need to create the sender for each message to send

Reply:
> EOL still missing

question: what is the purpose of adding an EOL to the file?

Reply:
https://stackoverflow.com/a/729795/6138465